### ISSUE TITLE
(🎁) should `N804` enforce `mcs` on metaclass `classmethod`s?

### ISSUE THREAD

**Issue by KotlinIsland** [astral-sh/ruff#13698]
```py
class M(type):
   def f(self):  # N804
       pass
   def m(cls):  # no error
       pass
   @classmethod
   def m(cls):  # no error
       pass
    @classmethod
    def c(mcs):  # N804
       pass
```
i would expect this to be `N804`, `no error`, `N804`, `no error`

**Comment by dhruvmanila** [comment]
There does seem to be some precedent for the "mcs" shorthand: https://grep.app/search?q=def%20%5B%5E(%5D+%5C(mcs%5B,%5C)%5D&regexp=true, @AlexWaygood any thoughts?

**Comment by eliegoudout** [comment]
Even though `mcs` itself maybe not be widely accepted, the fact that that it should be a one of `mcs`, `mcls`, `metacls`, <any other suggestion> to explicit difference in nature from `cls` (instance space in the case of metaclasses) seems more than reasonable (see https://github.com/astral-sh/ruff/issues/10656#issuecomment-2558202018).

I would love to see [https://github.com/astral-sh/ruff/issues/13698#issue-2577234229](https://github.com/astral-sh/ruff/issues/13698#issue-2577234229) (planned to be) fixed!

**Comment by eliegoudout** [comment]
Note that official library `abc` uses `mcls` for example:

https://github.com/python/cpython/blob/2a66dd33dfc0b845042da9bb54aaa4e890733f54/Lib/abc.py#L105-L108

**Comment by MichaReiser** [comment]
Related discussion about changing the rule not to enforce any naming convention for metaclass methods https://github.com/astral-sh/ruff/issues/6017
